src/shims/fs.rs

```diff
@@ -148,6 +152,13 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let fd = this.read_scalar(fd_op)?.to_i32()?;
 
         if let Some(handle) = this.machine.file_handler.handles.remove(&fd) {
+            // We avoid syncing the file if it was opened in read-only mode.
```

Comment:
Maybe add a link to some discussion explaining why.

Reply:
I added a brief explanation and a link to #999